src/commands.rs

```diff
@@ -36,13 +37,79 @@ pub enum EnvAction {
     Remove,
 }
 
+/// Define what strategy to use when choosing privilege scalation programs.
+#[allow(dead_code)]
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum ElevationStrategy {
+    Auto,
+    Prefer(OsString),
+    Force(&'static str),
+}
+
+impl ElevationStrategy {
+    pub fn resolve(&self) -> Result<OsString> {
+        match self {
+            ElevationStrategy::Auto => Self::choice(),
+            ElevationStrategy::Prefer(program) => which(program)
+                .map(std::path::PathBuf::into_os_string)
+                .or_else(|_| {
+                    let auto = Self::choice()?;
+                    warn!(
+                        "{} not found. Using {} instead",
+                        program.to_string_lossy(),
+                        auto.to_string_lossy()
+                    );
+                    Ok(auto)
+                }),
+            ElevationStrategy::Force(program) => Ok(program.into()),
+        }
+    }
+
+    /// Gets a path to a previlege elevation program based on what is available in the system.
+    ///
+    /// This funtion checks for the existence of common privilege elevation program names in
+    /// the `PATH` using the `which` crate and returns a Ok result with the `OsString` of the
+    /// path to the binary. In the case none of the checked programs are found a Err result is
+    /// returned.
+    ///
+    /// The search is done in this order:
+    ///
+    /// 1. `doas`
+    /// 1. `sudo`
+    /// 1. `run0`
+    /// 1. `pkexec`
+    ///
+    /// The logic for choosing this order is that a person with `doas` installed is more likely
+    /// to be using it as their main privilege elevation program.
+    /// `run0` and `pkexec` are preinstalled in any `NixOS` system with polkit support installed,
+    /// so they have been placed lower as it's easier to deactivate sudo than it is to remove
+    /// `run0`/`pkexec`
+    ///
+    /// # Returns
+    ///
+    /// * `Result<OsString>` - The absolute path to the privilege elevation program binary or an error if a
+    /// program can't be found.
+    fn choice() -> Result<OsString> {
+        const STRATEGIES: [&str; 4] = ["doas", "sudo", "run0", "pkexec"];
+
+        for strategy in STRATEGIES {
+            if let Ok(path) = which(strategy) {
+                debug!(?path, "{strategy} path found");
+                return Ok(path.into_os_string());
+            }
+        }
+
+        Err(eyre::eyre!("No elevation strategy found"))
```

Comment:
```suggestion
        Err(eyre::eyre!("No elevation strategy found. Checked: {}", STRATEGIES.join(", ")))
```

Or similar, for the sake of clarity.